File: cs_console/GuessingGame.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace cs_console
{
    class GuessingGame
    {
        static void Main(string[] args)
        {
            var random = new Random((Int32)DateTime.Now.Ticks);
            var rangeStart = 1;
            var rangeEnd = 10;
            var number = random.Next(rangeStart, rangeEnd);
            Console.WriteLine("Pick a number between {0} and {1}", rangeStart, rangeEnd);

            var notGuessed = true;
            var guessCount = 0;
            while (notGuessed)
            {
                var userEntered = Console.ReadLine();
                var numberGuessed = 0;
                if (Int32.TryParse(userEntered, out numberGuessed))
                {
                    guessCount += 1;
                    if (numberGuessed == number)
                    {
                        Console.WriteLine("{0} was the number. It took you {1} guesses", numberGuessed, guessCount);
                        notGuessed = false;
                    }
                    else
                    {
                        if (numberGuessed < rangeStart || numberGuessed > rangeEnd)
                        {
                            Console.WriteLine("That number is not between {0} and {1}", rangeStart, rangeEnd);
                            continue;
                        }
                        if (numberGuessed < number)
                        {
                            Console.WriteLine("That number is too low");
                            continue;
                        }
                        if (numberGuessed > number)
                        {
                            Console.WriteLine("That number is too high");
                            continue;
                        }
                    }
                }
                else
                {
                    Console.WriteLine("Invalid number. Try again.");
                }
            }
        }
    }
}

File: vb_console/GuessingGame.vb
Module GuessingGame

    Sub Main()
        Dim random As New Random(CType(DateAndTime.Now.Ticks Or &HFFFF0000, Int32))
        Dim rangeStart = 1
        Dim rangeEnd = 10
        Dim number = random.Next(rangeStart, rangeEnd)
        Console.WriteLine("Pick a number between {0} and {1}", rangeStart, rangeEnd)
        Dim notGuessed = True
        Dim guessCount = 0
        Do While notGuessed
            Dim userEntered = Console.ReadLine()
            Dim numberGuessed = 0
            If Int32.TryParse(userEntered, numberGuessed) Then
                guessCount += 1
                If numberGuessed = number Then
                    Console.WriteLine("{0} was the number. It took you {1} guesses", numberGuessed, guessCount)
                    notGuessed = False
                Else
                    If numberGuessed < rangeStart Or numberGuessed > rangeEnd Then
                        Console.WriteLine("That number is not between {0} and {1}", rangeStart, rangeEnd)
                        Continue Do
                    End If
                    If (numberGuessed < number) Then
                        Console.WriteLine("That number is too low")
                        Continue Do
                    End If
                    If (numberGuessed > number) Then

                        Console.WriteLine("That number is too high")
                        Continue Do
                    End If
                End If
            Else
                Console.WriteLine("Invalid number. Try again.")
            End If
        Loop
    End Sub

End Module
